src/Pim/Component/Catalog/Model/AbstractCompleteness.php

```diff
@@ -14,76 +18,83 @@
     /** @var int|string */
     protected $id;
 
+    /** @var ProductInterface */
+    protected $product;
+
     /** @var LocaleInterface */
     protected $locale;
 
     /** @var ChannelInterface */
     protected $channel;
 
     /** @var int */
-    protected $ratio = 100;
+    protected $ratio;
 
     /** @var int */
-    protected $missingCount = 0;
+    protected $missingCount;
 
     /** @var int */
-    protected $requiredCount = 0;
-
-    /** @var ProductInterface */
-    protected $product;
+    protected $requiredCount;
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getLocale()
-    {
-        return $this->locale;
-    }
+    /** @var Collection */
+    protected $missingAttributes;
 
     /**
-     * {@inheritdoc}
+     * @param ProductInterface $product
+     * @param ChannelInterface $channel
+     * @param LocaleInterface  $locale
+     * @param Collection       $missingAttributes
+     * @param int              $missingCount
+     * @param int              $requiredCount
      */
-    public function setLocale(LocaleInterface $locale)
-    {
+    public function __construct(
```

Comment:
Problem is it implies to find in the current completenesses the right completeness to update, which is harder than just replacing it.